src/wacks4square/wack.py

```diff
@@ -0,0 +1,83 @@
+import argparse
+import logging
+from datetime import datetime
+from pathlib import Path
+
+from dotenv import load_dotenv
+
+from wacks4square.square import Square
+from wacksbywarby.constants import WACK_ERROR_SENTINEL, SHIFT4SHOP_TIME_FORMAT
+from wacksbywarby.db import Wackabase
+from wacksbywarby.discord import Discord
+from wacksbywarby.wack import announce_new_sales
+
+DATABASE_DIR = "data/wack4square"
+
+load_dotenv()
+
+logger = logging.getLogger("wack4square")
+
+
+def main(db: Wackabase, dry=False):
+    try:
+        logger.info("TIME TO WACK")
+        logger.debug("Dry run: %s", dry)
+
+        discord = Discord(debug=dry)
+        square = Square(debug=dry)
+
+        last_timestamp = db.get_timestamp()
+        # stored in db with shift4shop format, but square requires RFC 3339 format so let's convert it
+        if last_timestamp:
+            last_timestamp = datetime.strptime(
+                last_timestamp, SHIFT4SHOP_TIME_FORMAT
+            ).isoformat()
+
+        last_num_sales = db.get_last_num_sales()
+        sales = square.get_sales_since_timestamp(timestamp=last_timestamp)
+        if not sales:
+            return
+
+        logger.info(f"last timestamp was {last_timestamp}")
+
+        # grab the current number of total sales
+        if last_num_sales:
+            current_num_sales = square.get_num_sales(
+                last_timestamp=last_timestamp, prev_num_sales=last_num_sales
+            )
+        else:
+            # backfill using slow method
+            current_num_sales = square.get_num_sales_slow()
+
+        logger.info(f"current num sales: {current_num_sales}")
+        announce_new_sales(discord, sales, current_num_sales, id_type="square")
+
+        # write out the most recent sale's date (results were sorted by closed at asc, so latest one is first one)
+        latest_sale_timestamp = sales[0].datetime
+        if latest_sale_timestamp:
+            db.write_timestamp(latest_sale_timestamp)
+            db.write_num_sales(current_num_sales)
+
+    except Exception as e:
+        logger.error("%s: %s", WACK_ERROR_SENTINEL, e)
+        raise
+
+
+if __name__ == "__main__":
```

Comment:
it's neat how this lines up so closely to wack4shop's pattern! if we were more industrious perhaps we'd make a factory pattern out of it...

Reply:
yeah..